### PR TITLE
[soul-admin] test: add test case for NacosDataChangedListener

### PR DESCRIPTION
**Pull request by cottom**
This PR is to add test case for `/soul/admin/listener/nacos/NacosDataChangedListener.java`

Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/en-us/docs/soul/contributor.html).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosMockConfigService.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dromara.soul.admin.listener.nacos;
+
+import com.alibaba.nacos.api.config.ConfigService;
+import com.alibaba.nacos.api.config.listener.Listener;
+import com.alibaba.nacos.api.exception.NacosException;
```
Comment (songyuequan) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosMockConfigService.java:22]: remove empty line

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * The TestCase for NacosDataChangedListener.
+ *
+ * @author cottom
+ * @date 2020/12/4
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class NacosDataChangedListenerTest {
```
Comment (songyuequan) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java:47]: add final on the class

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dromara.soul.admin.listener.nacos;
+
+import com.alibaba.nacos.api.config.ConfigService;
+import com.alibaba.nacos.api.exception.NacosException;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
```
Comment (songyuequan) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java:23]: remove empty line

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosMockConfigService.java (hunk 1 above)
Comment (cottom) [reply soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosMockConfigService.java:22]: what's the point?

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java (hunk 3 above)
Comment (cottom) [reply soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java:23]: same here.

**Comment by cottom** [comment]
@yu199195 PTAL

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/**
+ * The TestCase for NacosDataChangedListener.
+ *
+ * @author cottom
+ * @date 2020/12/4
```
Comment (dengliming) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java:44]: Remove `@date`

File: soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+        nacosDataChangedListener.onAppAuthChanged(
+                ImmutableList.of(appAuthData, appAuthData2), DataEventTypeEnum.CREATE);
+        Assert.assertEquals(
+                configService.getConfig(AUTH_DATA_ID, GROUP, 100),
+                GsonUtils.getInstance()
+                        .toJson(ImmutableMap.of(mockAppKey2, appAuthData2, mockAppKey, appAuthData)));
+        nacosDataChangedListener.onAppAuthChanged(
+                ImmutableList.of(appAuthData), DataEventTypeEnum.DELETE);
+        Assert.assertEquals(
+                configService.getConfig(AUTH_DATA_ID, GROUP, 100),
```
Comment (dengliming) [inline soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java:88]: Here it's easy to get confused. Why does the config still exist when we've deleted it. You should add more comment about what `NacosMockConfigService`  it really does.

Comment (cottom) [reply soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java:88]: Why confused? We have `appAuthData ` and `appAuthData2 ` in the config. Now we have only delete the `appAuthData`, the left `appAuthData2` is as a matter of course @dengliming

Comment (dengliming) [reply soul-admin/src/test/java/org/dromara/soul/admin/listener/nacos/NacosDataChangedListenerTest.java:88]: em my bad.

**Review by dengliming — COMMENTED** [review]
Looks good. Please just remove `@date`. and we can merge. Thanks.

**Review by tydhot — APPROVED** [review]

**Pull request merged**